modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -992,6 +997,30 @@ private ObjectRelationship _updateObjectRelationship(
 		return objectRelationshipPersistence.update(objectRelationship);
 	}
 
+	private void _validateEdge(
+			boolean edge, long objectDefinitionId1, long objectDefinitionId2,
+			String type)
+		throws ObjectRelationshipEdgeException {
+
+		if (!edge) {
+			return;
+		}
+
+		if (!Objects.equals(
+				type, ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {
+
+			throw new ObjectRelationshipEdgeException(
+				"Object relationship must be one to many to be an edge of a " +
+					"root context");
+		}
+
+		if (objectDefinitionId1 == objectDefinitionId2) {
+			throw new ObjectRelationshipEdgeException(
+				"Object relationship must not be a self-relationship to be " +
+					"an edge of a root context");
+		}
+	}
```

Comment:
@gabrielwas here we can pass the objectRelationship as parameter, so we can do:

```suggestion
	private void _validateEdge(boolean edge, ObjectRelationship objectRelationship)
		throws ObjectRelationshipEdgeException {

		if (!edge) {
			return;
		}

		if (!Objects.equals(
				objectRelationship.getType(), ObjectRelationshipConstants.TYPE_ONE_TO_MANY)) {

			throw new ObjectRelationshipEdgeException(
				"Object relationship must be one to many to be an edge of a " +
					"root context");
		}

		if (objectRelationship.isSelf()) {
			throw new ObjectRelationshipEdgeException(
				"Object relationship must not be a self-relationship to be " +
					"an edge of a root context");
		}
	}
```

there is another validation that I believe we can do here, we need to check if the father is a unmodifiable system object, if so, we need to throw an exception.

Thanks!

Reply:
I did it at first, but then I notice the pattern of the validation methods is passing the parameters (and not the major entity).